### PR TITLE
fix: fix observer version check and add sync in compute/compactor node

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by yezizp2012** [comment]
@Li0k can we unified Implementation for `CompactorObserverNode` and `ComputeObserverNode`? Because they look exactly the same.

**Review by Li0k — APPROVED** [review]
LGTM！ Thanks for this PR

**Comment by Li0k** [comment]
> @Li0k can we unified Implementation for `CompactorObserverNode` and `ComputeObserverNode`? Because they look exactly the same.

for this case (only for FilterKeyExtrator) yes, but i am not sure that `ComputeNode` has other needs, like that `ComputeNode` restore catalog for other features. Therefore, I provide the indenpendent observer_manager for `ComputeNode`. (now they are exactly the same)

File: src/storage/hummock_sdk/src/filter_key_extractor.rs [hunk 1]
```diff
@@ -233,6 +245,15 @@ impl FilterKeyExtractorManagerInner {
         self.notify.notify_waiters();
     }
 
+    fn sync(&self, filter_key_extractor_map: HashMap<u32, Arc<FilterKeyExtractorImpl>>) {
+        let mut guard = self.table_id_to_filter_key_extractor.write();
+        guard.clear();
+        for (table_id, filter_key_extractor) in filter_key_extractor_map {
+            guard.insert(table_id, filter_key_extractor);
+        }
```
Comment (HuaHuaY) [inline src/storage/hummock_sdk/src/filter_key_extractor.rs:253]:
```suggestion
        guard.extend(filter_key_extractor_map);
```

**Review by HuaHuaY — APPROVED** [review]
LGTM

**Pull request merged**